pkg/cache/internal/cache_reader_test.go

```diff
@@ -0,0 +1,77 @@
+package internal
```

Comment:
Why the benchmark? This isn't suited if people care about performance, its just for convenience. If someome needs performance, they should be creating one index for all things they want to index, i.E. in your testexample a label+restart policy index

Reply:
Try to keep in mind there is a potential performance issue with multiple indexes. By tracking upstream `ByIndexes`, if upstream implement and we'll switch to standard `ByIndexes`.

Reply:
Yes I see that - Which is why I think this benchmark is pointless. Anyone who cares about perf should not be using this.

Reply:
OK, I will delete it.